File: 4-median-of-two-sorted-arrays/median-of-two-sorted-arrays.py
class Solution:
    def findMedianSortedArrays(self, nums1: List[int], nums2: List[int]) -> float:
        p1, p2 = 0, 0
        m, n = len(nums1), len(nums2)
        
        ans = [0] * (int((m+n)/2) + 1) 
        # print(ans)

        i = 0 
        while i<len(ans):
            if p1 < m and p2 < n:
                if nums1[p1] < nums2[p2]:
                    ans[i] = nums1[p1]
                    i+=1
                    p1+=1
                    # print(ans)

                else:
                    ans[i] = nums2[p2]
                    i+=1
                    p2+=1
                    # print(ans)

            elif p1 == m:
                ans[i] = nums2[p2]
                i+=1
                p2+=1

            else:
                ans[i] = nums1[p1]
                i+=1
                p1+=1
        # print(ans)
        if (m+n)%2 == 0:
            return (ans[-1] + ans[-2])/2

        else:
            return ans[-1]

            

File: 4-median-of-two-sorted-arrays/median-of-two-sorted-arrays.cpp
class Solution {
public:
    double findMedianSortedArrays(vector<int>& nums1, vector<int>& nums2) {
        vector<int> nums3;
        int u = 0;
        int v = 0;

        while (u < nums1.size() && v < nums2.size()){
            if(nums1[u] <= nums2[v]){
                nums3.push_back(nums1[u]);
                u++;
            }

            else if (nums2[v] < nums1[u]){
                nums3.push_back(nums2[v]);
                v++;
            }
        }

        while (u < nums1.size()){
            nums3.push_back(nums1[u]);
            u++;
        }

        while (v < nums2.size()){
            nums3.push_back(nums2[v]);
            v++;
        }

        for(int i = 0; i <nums3.size(); i++){
            std::cout<<nums3[i]<<std::endl;
        }

        int n = nums3.size();
        if (n%2 != 0) return nums3[n/2];
        else return (nums3[n/2 - 1] + nums3[n/2])/2.0;
        // return 2.3;

    }
};